### PR TITLE
Track asynchronous operations running in ServiceHub

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs [hunk 1]
```diff
@@ -145,7 +145,7 @@ private static void DisposeExportProvider(ExportProvider? exportProvider)
 
                     try
                     {
-                        var waiter = ((AsynchronousOperationListenerProvider)listenerProvider).WaitAllDispatcherOperationAndTasksAsync();
+                        var waiter = ((AsynchronousOperationListenerProvider)listenerProvider).WaitAllDispatcherOperationAndTasksAsync(null);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs:148]: nit: named arg?

File: src/Workspaces/Remote/ServiceHub/Services/Host/RemoteAsynchronousOperationListenerService.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        public ValueTask<bool> IsCompletedAsync(ImmutableArray<string> featureNames, CancellationToken cancellationToken)
+        {
+            return RunServiceAsync(cancellationToken =>
+            {
+                var workspace = GetWorkspace();
+                var exportProvider = (IMefHostExportProvider)workspace.Services.HostServices;
+                var listenerProvider = exportProvider.GetExports<AsynchronousOperationListenerProvider>().Single().Value;
+
+                return new ValueTask<bool>(!listenerProvider.HasPendingWaiter(featureNames.ToArray()));
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Services/Host/RemoteAsynchronousOperationListenerService.cs:38]: pity that hasPendingWaiter takes an array and can't handle an immutablearray.  cleanup for future maybe

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/Remote/ServiceHub/Services/Host/RemoteAsynchronousOperationListenerService.cs (hunk 2 above)
Comment (sharwell) [reply src/Workspaces/Remote/ServiceHub/Services/Host/RemoteAsynchronousOperationListenerService.cs:38]: I'm not going to touch that signature while we still have unrestricted IVTs.

File: src/Workspaces/CoreTestUtilities/AsynchronousOperationListenerExtensions.cs [hunk 3]
```diff
@@ -2,15 +2,20 @@
...
 
 namespace Roslyn.Test.Utilities
 {
     public static class AsynchronousOperationListenerExtensions
     {
-        internal static Task WaitAllDispatcherOperationAndTasksAsync(this IAsynchronousOperationListenerProvider provider, params string[] featureNames)
-            => ((AsynchronousOperationListenerProvider)provider).WaitAllAsync(featureNames);
+        internal static async Task WaitAllDispatcherOperationAndTasksAsync(this IAsynchronousOperationListenerProvider provider, Workspace? workspace, params string[] featureNames)
+        {
+            await ((AsynchronousOperationListenerProvider)provider).WaitAllAsync(workspace, featureNames).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Workspaces/CoreTestUtilities/AsynchronousOperationListenerExtensions.cs:17]: In the past I've thought about cleaning this up via a new interface, but everything seems happy doing it this way so I've ended up leaving it.

File: src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs (hunk 1 above)
Comment (sharwell) [reply src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs]: ➡️ Done and commented

**Pull request merged**